tests/integration/targets/iosxr_ospfv2/vars/main.yaml

```diff
@@ -0,0 +1,356 @@
+
+  commands:
+    - router ospf 27
+    - area 20 authentication keychain ansi11393
+    - area 20 default-cost 2
+    - area 20 cost 2
+
+  after:
+    processes:
+      - areas:
+          - area_id: '11'
+            cost: 11
+            default_cost: 5
+          - area_id: '22'
+            default_cost: 6
+        authentication:
+          keychain: 'ansible_test1102'
+        process_id: '10'
+      - adjacency_stagger:
+          max_adjacency: 20
+          min_adjacency: 10
+        authentication:
+          message_digest:
+            keychain: 'ansible1101pass'
+        process_id: '26'
+      - areas:
+          - area_id: '10'
+            authentication:
+              keychain: 'ansi11393'
+            hello_interval: 2
+          - area_id: '20'
+            authentication:
+              keychain: 'ansi11393'
+            cost: 2
+            default_cost: 2
+        process_id: '27'
+      - areas:
+          - area_id: '11'
+            default_cost: 5
+          - area_id: '22'
+            default_cost: 6
+        cost: 2
+        dead_interval: 2
+        default_metric: 10
+        demand_circuit: 'enable'
+        external_out: 'disable'
+        flood_reduction: 'disable'
+        hello_interval: 1
+        mtu_ignore: 'enable'
+        packet_size: 577
+        passive: 'disable'
+        priority: 1
+        process_id: '30'
+        retransmit_interval: 2
+        router_id: '2.2.2.2'
+        summary_in: 'enable'
+        transmit_delay: 2
+        weight: 2
+
+overridden:
+  commands:
+    - router ospf 10
+    - no authentication keychain ansible_test1102
+    - no area 11 default-cost 5
+    - no area 11 cost 11
+    - no area 22 default-cost 6
+    - router ospf 30
+    - no cost 2
+    - no weight 2
+    - no passive disable
+    - no priority 1
+    - no flood-reduction disable
+    - no default-metric 10
+    - no router-id 2.2.2.2
+    - no demand-circuit enable
+    - no packet-size 577
+    - no transmit-delay 2
+    - no summary-in enable
+    - no external-out disable
+    - no dead-interval 2
+    - no hello-interval 1
+    - no retransmit-interval 2
+    - no mtu-ignore enable
+    - no area 11 default-cost 5
+    - no area 22 default-cost 6
+    - router ospf 27
+    - area 20 authentication keychain ansi11393
+    - area 20 default-cost 2
+    - area 20 cost 2
+  after:
+    processes:
+      - process_id: '10'
+      - adjacency_stagger:
+          max_adjacency: 20
+          min_adjacency: 10
+        authentication:
+          message_digest:
+            keychain: 'ansible1101pass'
+        process_id: '26'
+      - areas:
+          - area_id: '10'
+            authentication:
+              keychain: 'ansi11393'
+            hello_interval: 2
+          - area_id: '20'
+            authentication:
+              keychain: 'ansi11393'
+            cost: 2
+            default_cost: 2
+        process_id: '27'
+      - process_id: '30'
+deleted:
+  commands:
+    - router ospf 10
```

Comment:
Shouldn't the delete operation, delete the given ospf process by 'no router ospf 10' ?

Reply:
keeping the complexity of OSPF in iosxr mind and as per the new discussed approach of having separate OSPF vrf support RM, It makes sense not to delete
the ospfv2 processes, so delete or overridden would impact the processes to the scope of attributes. In future, we might want to have separate ospf_vrf(to manage vrf support) and ospf_global (to globally handle OSPF process complete removal)

Reply:
@rohitthakur2590 Are we having a separate OSPF VRF module? I thought the partition was simply based on v2 and v3. I agree with @GomathiselviS here, we should negate the process directly with state deleted or superfluous processes with state overridden.

Reply:
@NilashishC  yeah So iosxr vrf has almost the same attributes as OSPF process upto the nested levels, it was getting too difficult to maintain, so we have decided to have separate ospf_vrf module.
Having a separate vrf module means we can't remove the OSPF processes completely and which makes sense as vrf is part of the OSPF process. So, for now its decided to have ospf_v2 process rm, and in future, we will have separate ospf_vrf rm or we might figure out a way for the vrf key in the global module to actually use the vrf module in the background.

SO  basically for us its better to make customer understand why we are not removing ospf process/vrf completely than to purge vrf/processes altogether here.